### PR TITLE
New package: libnma-1.8.30; network-manager-applet: update to 1.18.0 and add libappindicator support

### PR DESCRIPTION
**Pull request by cinerea0**
This commit adds support for `nm-applet`'s `--indicator` flag, which allows the applet to run in trays that use the AppIndicator protocol.

### CODE REVIEW

**Comment by cinerea0** [comment]
Pinging @Gottox because it's his package.

**Comment by sgn** [comment]
Should we also bump to v1.18.0? I don't use this package, so I don't really know.

**Comment by cinerea0** [comment]
I attempted to build v1.18.0, but ran into the following build error:

> configure: error: Package requirements (libnma >= 1.8.27) were not met

That file is provided by the `libnm-gtk` package, which is a subpackage of `network-manager-applet`. xbps-src appears to be unable to resolve this circular dependency.

**Comment by CameronNemo** [comment]
they must have split the source package https://gitlab.gnome.org/GNOME/libnma

**Comment by cinerea0** [comment]
Gotcha. What would be the correct order to perform these changes in? My first thought would be:

1. Package `libnma`(separate PR)

2. Change the `network-manager-applet` template to use `libnma` and get rid of the subpackages (this PR)

3. Revbump the packages you listed with updated dependencies (separate PR)

**Comment by CameronNemo** [comment]
all in one PR, separate commits for each package.

File: srcpkgs/libnma/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'libnma'
+pkgname=libnma
+version=1.8.30
+revision=1
+build_style=meson
+build_helper="gir"
```
Comment (CameronNemo) [inline srcpkgs/libnma/template:6]: you will need the "qemu" build helper too.

File: srcpkgs/libnma/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+hostmakedepends="cmake gettext glib-devel gtk-doc pkg-config vala"
```
Comment (CameronNemo) [inline srcpkgs/libnma/template:7]: are you sure you need cmake given this is a meson-built package?

File: srcpkgs/libnma/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+short_desc="NetworkManager panel applet for GNOME - runtime library"
+maintainer="cinerea0 <cinerea0@protonmail.com>"
+license="GPL-2.0-only"
+homepage="https://gitlab.gnome.org/GNOME/libnma"
+distfiles="https://gitlab.gnome.org/GNOME/libnma/-/archive/${version}/libnma-${version}.tar.gz"
+checksum=c874d83dd59468b68c79ff20e8f5a689a009c89b618e89b5939fbdc81776fbf5
+
+# Package build options
+build_options="gir"
+build_options_default="gir"
```
Comment (CameronNemo) [inline srcpkgs/libnma/template:20]: Typically we only specify options when something about a package is actually optional. In this case it always needs gir. Good rule of thumb is that a build option should actually change the behavior of the build, ejemplos gratis: dependencies, arguments passed to the build system.

File: srcpkgs/libnma/template [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+revision=1
+build_style=meson
+build_helper="gir"
+hostmakedepends="cmake gettext glib-devel gtk-doc pkg-config vala"
+makedepends="gcr-devel gtk+3-devel iso-codes
+ mobile-broadband-provider-info NetworkManager-devel"
+depends="NetworkManager iso-codes"
+short_desc="NetworkManager panel applet for GNOME - runtime library"
+maintainer="cinerea0 <cinerea0@protonmail.com>"
+license="GPL-2.0-only"
```
Comment (CameronNemo) [inline srcpkgs/libnma/template:13]: Looks like a non-trivial amount of the project is LGPL, so change this line to:

`license="GPL-2.0-only, LGPL-2.1-or-later"`

**Review by CameronNemo — COMMENTED** [review]
Looks like there is some issue with cross that is related to gobject introspection but not covered by our current gir handling. We could try to handle the gtkdoc-scanobj call with the gir build helper, but it would be easier to just use qemu.

File: srcpkgs/libnma/template (hunk 3 above)
Comment (cinerea0) [reply srcpkgs/libnma/template]: Gotcha, I've removed those lines.

File: srcpkgs/libnma/template (hunk 4 above)
Comment (cinerea0) [reply srcpkgs/libnma/template]: Thank you, I've changed the license like you said.

File: srcpkgs/libnma/template (hunk 2 above)
Comment (cinerea0) [reply srcpkgs/libnma/template]: When I was testing building the package locally I needed it. However, after my most recent test it seems like I don't. I've removed it from my most recent push and it hasn't introduced new errors.

File: srcpkgs/libnma/template (hunk 1 above)
Comment (cinerea0) [reply srcpkgs/libnma/template]: I changed the `build_helper` value to `"gir qemu"`, but I'm still getting the same errors as before on the ARM architectures.

**Comment by cinerea0** [comment]
It finally worked! Alright, `libnma` down, `network-manager-applet` is up next.

File: srcpkgs/libnma/template [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+if [ "$CROSS_BUILD" ]; then
+	configure_args+="-Dgtk_doc=false"
+	hostmakedepends+=" libnma"
+fi
+
+post_install() {
+	if [ "$CROSS_BUILD" ]; then
+		vmkdir usr/share/gtk-doc/html 0755
+		vcopy /usr/share/gtk-doc/html/libnma usr/share/gtk-doc/html
+		vmove usr/lib/girepository-1.0
```
Comment (CameronNemo) [inline srcpkgs/libnma/template:27]: why is this only being installed for cross builds? should it not just always be installed?

Comment (cinerea0) [reply srcpkgs/libnma/template:27]: That last line is a mistake and doesn't need to be there. As for the first two, the package fails to build documentation properly only on ARM architectures. Those lines install files that are only missing when built on ARM.

Comment (CameronNemo) [reply srcpkgs/libnma/template:27]: yeah I was following the convo in #xbps regarding the docs. the gir line is what I was referring to.

Comment (cinerea0) [reply srcpkgs/libnma/template:27]: Yeah, that'll be removed on my next push.

**Comment by cinerea0** [comment]
In the original `network-manager-applet` template, one of the configure options is "--disable-static". Does anyone know what option in the following file would most likely correspond to enabling/disabling static compilation?

```
option('appindicator', type: 'string', value: 'yes', description: 'Build with lib(ayatana-)appindicator support in addition to xembed systray support (no|yes|auto|ayatana|ubuntu). Default is "yes", "yes" is an alias for "auto".')
option('wwan', type: 'boolean', value: true, description: 'Enable WWAN support.')
option('selinux', type: 'boolean', value: true, description: 'Enable support for adjusting SELinux labels in configuration editor.')
option('team', type: 'boolean', value: true, description: 'Enable team configuration editor.')
option('more_asserts', type: 'string', value: '0', description: 'Enable more assertions for debugging.')
option('ld_gc', type: 'boolean', value: true, description: 'Enable garbage collection of unused symbols on linking')
```

**Comment by CameronNemo** [comment]
There is none from what I can tell. It would apply to the library, libnma, if it existed.

**Comment by cinerea0** [comment]
> There is none from what I can tell. It would apply to the library, libnma, if it existed.

I checked libnma's options, and it doesn't exist. It's probably fine to take that option out then.

**Comment by cinerea0** [comment]
Upon trying to revbump and build `NetworkManager-vpnc` with the new `libnma` dependency, I get the following error:

```
checking for LIBNMA... no
configure: error: Package requirements (libnma >= 1.2.0) were not met:

Package 'libnm', required by 'libnma', not found
```
However, `libnm` is in the makedepends. What's going wrong here? Do I need to be installing more files from `libnma`, like a `libnma-devel`?

**Comment by CameronNemo** [comment]
@cinerea0 check the last two commits here: https://github.com/CameronNemo/void-packages/commits/c0nma

Some notes on what I changed:

* using GNOME_SITE in distfiles, we never get the tarballs from the GNOME gitlab
* adjusting the common/shlibs file
* adding a libnma-devel subpackage -- when packaging libraries you almost always need a subpackage for the development headers
* adjusting network-manager-applet to build against the new libnma

All that is left from my perspective is revbumping the packages I mentioned above:

```
NetworkManager-l2tp-1.8.2_1
NetworkManager-openvpn-1.8.10_2
NetworkManager-pptp-1.2.8_3
NetworkManager-strongswan-1.4.5_2
NetworkManager-vpnc-1.2.6_3
cinnamon-control-center-4.6.2_1
gnome-control-center-3.34.1_1
gnome-initial-setup-3.34.1_2
switchboard-plug-network-2.1.4_1
```

On each of these packages, you will need to replace the `network-manager-applet-devel` dependency with `libnma-devel`.

**Comment by cinerea0** [comment]
After performing these changes, I'm still getting the error above when trying to build `NetworkManager-vpnc` with `libnma` in makedepends. When switching to `libnma-devel`, I get the following error:

```
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-bar-code-widget.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-bar-code.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-cert-chooser.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-mobile-providers.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-mobile-wizard.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ui-utils.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-version.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-vpn-password-dialog.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-wifi-dialog.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-802-1x.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-dynamic-wep.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-leap.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-sae.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-wep-key.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-wpa-eap.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws-wpa-psk.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/include/libnma/nma-ws.h' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/lib/pkgconfig/libnma.pc' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gir-1.0/NMA-1.0.gir' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/NMABarCode.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/NMACertChooser.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/NMAMobileProvidersDatabase.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/NMAMobileWizard.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/NMAVpnPasswordDialog.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/NMAWifiDialog.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/annotation-glossary.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/api-index-1.8.0.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/api-index-full.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/api-reference.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/home.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/index.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/left-insensitive.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/left.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/libnma-nma-ui-utils.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/libnma.devhelp2' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/object-tree.html' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/right-insensitive.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/right.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/style.css' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/up-insensitive.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/gtk-doc/html/libnma/up.png' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/vala/vapi/libnma.deps' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/share/vala/vapi/libnma.vapi' already installed by package libnma-1.8.30_1.
ERROR: libnma-devel-1.8.30_1: file `/usr/lib/libnma.so' already installed by package libnma-1.8.30_1.
Transaction failed! see above for errors.
```

**Comment by CameronNemo** [comment]
Rebuild libnma with the force flag: `./xbps-src -f pkg libnma`. I moved the docs from the regular package to the devel package.

**Comment by cinerea0** [comment]
@CameronNemo Yep, that worked. I'll refresh the travis build in the morning so every subsequent push doesn't lead to a failed build just because it doesn't recognize `libnma`. Final thing I've come across tonight is a new error when building `NetworkManager-vpnc` with `libnma-devel`:

```
checking for LIBNM_GTK... no
configure: error: Package requirements (libnm-gtk >= 1.2.0) were not met:

No package 'libnm-gtk' found
```

Since we're getting rid of `libnm-gtk` since everything I've seen makes it seem like the library is deprecated, I'm confused why this package requires it.

**Comment by CameronNemo** [comment]
I would just:

* file an issue upstream https://gitlab.gnome.org/GNOME/NetworkManager-vpnc . say "FTBFS with recent versions of libnma"
* mark the package broken, linking to the issue you filed

**Comment by CameronNemo** [comment]
@Gottox your name is all over these packages, you might want to take a look.

**Comment by cinerea0** [comment]
@CameronNemo The most recent push failed because one of the targets could not fetch the distfile; this problem did not occur on any of the other targets. What should I do?

**Comment by CameronNemo** [comment]
Did you try pushing again? Could have just been a blip.

**Comment by cinerea0** [comment]
Yep, that worked. I'll hopefully finish up the last of the package updates today.

**Comment by CameronNemo** [comment]
For the switchboard plug, you probably need to add `build_helper="gir"`

**Comment by cinerea0** [comment]
Still failing on ARM even after adding `build_helper=gir`, unfortunately.

**Comment by CameronNemo** [comment]
@cinerea0 then just mark it nocross for now, we will have to figure it out later

**Comment by cinerea0** [comment]
Seriously, two transient resolver errors in a row? This is crazy-making.

**Comment by cinerea0** [comment]
Three times in a row! What am I even supposed to do about this?

**Comment by cinerea0** [comment]
You know what? I know that it builds properly on all architectures because the DNS failure keeps happening on different arches. I'll be moving on to the last one shortly.

**Comment by CameronNemo** [comment]
the vpnc plugin should not need libnma-devel... see the diff I included in [my comment above](https://github.com/void-linux/void-packages/pull/24497#issuecomment-704008662)

**Comment by cinerea0** [comment]
@CameronNemo I discovered that the error I was encountering before was due to a typo in the configure_args line, not because of anything wrong with the vpnc plugin itself. As such, the version I pushed with `libnma-devel` should build properly on all archs, at least if travis would stop having connectivity problems.

**Comment by cinerea0** [comment]
Alright, here's a screenshot of the last build. The two builds that failed did so because of a DNS failure on Travis's end. If the force-push I'm about to do results in non-overlapping failing builds (or even a completely successful build), I'll mark this as ready for review so it can hopefully be merged.
Screenshot_2020-10-10 Build #39844 - void-linux void-packages - Travis CI

**Comment by cinerea0** [comment]
New error with the new push:

> ERROR: gnome-control-center-3.38.1_1: failed to install 'ModemManager-devel-1.12.10_1' dependency!

Not even sure how to solve this, given that this package exists in the repos.

**Comment by cinerea0** [comment]
The most recent force-push only failed on armv6 due to a DNS error which, as you can see above, does not overlap with the previous error. As such, I am marking this PR ready for review with the hope that it will be merged. I would also request that whoever merges this:

1. Removes `network-manager-applet-devel` and `libnm-gtk` from the repository index.
2. Checks out #24776 to see if it can also be merged.

**Comment by cinerea0** [comment]
@Vaelatern if you have time, could you take a look at this PR?

**Comment by Piraty** [comment]
we don't want merge commits, rebase instead.

**Comment by cinerea0** [comment]
@Piraty Fixed! I would prefer to not have to do it again before this gets merged, though.

**Pull request merged**